### PR TITLE
[BACKLOG-35397] Fix integration tests for cpf project to run in both …

### PR DESCRIPTION
**Pull request by smmribeiro**
…Java 11 and Java 8

Updated unit tests in order to be more compliant with the JUnit version were now using (there were JUnit 3.0 tests, deprecated imports, etc.).

@bcostahitachivantara @renato-s @andreramos89

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **72**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.pentaho.net/job/projects/job/cpf-plugin/job/PR-168/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 39s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **72**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.pentaho.net/job/projects/job/cpf-plugin/job/PR-168/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by ricksjames — APPROVED** [review]

File: core/src/test/java/pt/webdetails/cpf/packager/MinificationPackageTest.java [hunk 1]
```diff
@@ -1,145 +0,0 @@
-package pt.webdetails.cpf.packager;
```
Comment (bcostahitachivantara) [inline core/src/test/java/pt/webdetails/cpf/packager/MinificationPackageTest.java:1]: Why did you remove this class?

File: core/src/test/java/pt/webdetails/cpf/repository/VfsRepositoryTest.java [hunk 2]
```diff
@@ -62,14 +68,17 @@ public void testFolderCreation() throws IOException {
     assertTrue( doNothing );
   }
 
-  public void testResourceExists() throws IOException {
+  @Test
+  public void testResourceExists() throws Exception {
     repository.createFolder( "testFolderCreation" );
     boolean exists = repository.resourceExists( "testFolderCreation" );
+    assertTrue( exists );
     boolean notExists = repository.resourceExists( "IAmNotAFolder!" );
```
Comment (bcostahitachivantara) [inline core/src/test/java/pt/webdetails/cpf/repository/VfsRepositoryTest.java:76]: This variable has a ambiguous name because has the name notExists but the value is false instead of true.

File: pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java [hunk 3]
```diff
@@ -31,69 +29,27 @@
...
   public void testGetExtensionOnlyKnownInPlatform() {
 
-    String actualOutcome = StringUtils.EMPTY;
-
-    try {
-
-      actualOutcome = MimeTypeHandler.getMimeTypeFromExtension( EXTENSION_ONLY_KNOW_IN_PLATFORM );
+    String actualOutcome = MimeTypeHandler.getMimeTypeFromExtension( EXTENSION_ONLY_KNOW_IN_PLATFORM );
 
-      log( "testGetExtensionOnlyKnownInPlatform()", EXTENSION_ONLY_KNOW_IN_PLATFORM, EXPECTED_OUTCOME_IN_PLATFORM,
```
Comment (bcostahitachivantara) [inline pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java:45]: Why did you remove this log?

File: pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java [hunk 4]
```diff
@@ -31,69 +29,27 @@
...
   public void testGetExtensionOnlyKnownInCpf() {
 
-    String actualOutcome = StringUtils.EMPTY;
-
-    try {
-
-      actualOutcome = MimeTypeHandler.getMimeTypeFromExtension( EXTENSION_ONLY_KNOW_IN_CPF );
+    String actualOutcome = MimeTypeHandler.getMimeTypeFromExtension( EXTENSION_ONLY_KNOW_IN_CPF );
 
-      log( "testGetExtensionOnlyKnownInCpf()", EXTENSION_ONLY_KNOW_IN_CPF, EXPECTED_OUTCOME_IN_CPF, actualOutcome );
```
Comment (bcostahitachivantara) [inline pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java:65]: why did you remove this log?

File: pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java [hunk 5]
```diff
@@ -31,69 +29,27 @@
...
   @Test
   public void testGetExtensionUnknownByAll() {
 
-    String actualOutcome = StringUtils.EMPTY;
-
-    try {
-
-      actualOutcome = MimeTypeHandler.getMimeTypeFromExtension( EXTENSION_UNKNOWN_BY_ALL );
-
-      log( "testGetExtensionUnknownByAll()", EXTENSION_UNKNOWN_BY_ALL, EXPECTED_OUTCOME_UNKNOWN_BY_ALL,
```
Comment (bcostahitachivantara) [inline pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java:84]: why did you remove this log?

File: pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java [hunk 6]
```diff
@@ -31,69 +29,27 @@
...
-    } catch ( Throwable t ) {
-      logger.error( t );
-      Assert.fail();
-    }
-
-    Assert.assertTrue( EXPECTED_OUTCOME_UNKNOWN_BY_ALL.equalsIgnoreCase( actualOutcome ) );
-  }
+    String actualOutcome = MimeTypeHandler.getMimeTypeFromExtension( EXTENSION_UNKNOWN_BY_ALL );
 
-  private void log( String methodName, String testExtension, String expectedOutcome, String actualOutcome ) {
```
Comment (bcostahitachivantara) [inline pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java:95]: why did you remove this log method? (related with the previous questions)

File: pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java [hunk 7]
```diff
@@ -78,12 +80,10 @@ public void testGetOlapCubes() throws Exception {
...
-    try {
-      result = olapUtils.getOlapCubes();
-    } catch ( JSONException e ) {
-      Assert.fail( e.getMessage() );
-    }
-    Assert.assertTrue( jsonEquals( expectedResult, result.toString() ) );
+
+    result = olapUtils.getOlapCubes();
+
+    assertTrue( jsonEquals( expectedResult, result.toString() ) );
```
Comment (bcostahitachivantara) [inline pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java:86]: Can you use here the assertEquals?

File: pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java [hunk 8]
```diff
@@ -102,15 +102,10 @@ public void testGetCubeStructure() throws Exception {
...
-      result =
-          olapUtils.getCubeStructure( "", "", "" );
-    } catch ( JSONException e ) {
-      Assert.fail( e.getMessage() );
-    }
-    Assert.assertTrue( jsonEquals( expectedResult, result.toString() ) );
-  }
+    result = olapUtils.getCubeStructure( "", "", "" );
 
+    assertTrue( jsonEquals( expectedResult, result.toString() ) );
```
Comment (bcostahitachivantara) [inline pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java:107]: Can you use here the assertEquals?

File: pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java [hunk 9]
```diff
@@ -125,12 +120,10 @@ public void testGetMeasures() throws Exception {
...
-    try {
-      result = olapUtils.getMeasures( mockedConnection, "" );
-    } catch ( JSONException e ) {
-      Assert.fail( e.getMessage() );
-    }
-    Assert.assertTrue( jsonEquals( expectedResult, result.toString() ) );
+
+    result = olapUtils.getMeasures( mockedConnection, "" );
+
+    assertTrue( jsonEquals( expectedResult, result.toString() ) );
```
Comment (bcostahitachivantara) [inline pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java:126]: Can you use here the assertEquals?

File: pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java [hunk 10]
```diff
@@ -146,12 +139,10 @@ public void testGetLevelMembersStructure() throws Exception {
...
-    try {
-      result = olapUtils.getLevelMembersStructure( "", "", "", "" );
-    } catch ( JSONException e ) {
-      Assert.fail( e.getMessage() );
-    }
-    Assert.assertTrue( jsonEquals( expectedResult, result.toString() ) );
+
+    result = olapUtils.getLevelMembersStructure( "", "", "", "" );
+
+    assertTrue( jsonEquals( expectedResult, result.toString() ) );
```
Comment (bcostahitachivantara) [inline pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java:145]: Can you use here the assertEquals?

File: pentaho/src/test/java/pt/webdetails/cpf/persistence/PersistenceTest.java [hunk 11]
```diff
@@ -141,13 +127,10 @@ public void testEscaping() {
         "\r\n            \"likeJson\": \"[1,2,3]\",\r\n            \"spaces\":  \"value with spaces\"\r\n        " +
         "}\r\n    }\r\n}";
     doc.fromJSON( s );
-    Assert.assertEquals( doc.field( "deep[deeper][quotes]" ), "\"\",\"oops\":\"123\"" );
+    assertEquals( "\"\",\"oops\":\"123\"", doc.field( "deep[deeper][quotes]" ) );
```
Comment (bcostahitachivantara) [inline pentaho/src/test/java/pt/webdetails/cpf/persistence/PersistenceTest.java:130]: Can you put the expected result into a variable? (just for readability)

**Review by bcostahitachivantara — CHANGES_REQUESTED** [review]

File: core/src/test/java/pt/webdetails/cpf/localization/test/PluginWriteAccess.java [hunk 12]
```diff
@@ -54,7 +54,7 @@
   }
 
   @Override public boolean fileExists( String path ) {
-    return storedFiles != null && storedFiles.contains( new BasicFile( path ) );
+    return storedFiles.contains( new BasicFile( path ) );
```
[the remark this answers is not in the document]
Comment (smmribeiro) [reply core/src/test/java/pt/webdetails/cpf/localization/test/PluginWriteAccess.java:57]: It is a final variable initialized on line 26, so it is never null.

File: pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java (hunk 9 above)
Comment (smmribeiro) [reply pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java:126]: No, because the comparison involves some intermediate transformations.

File: pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java (hunk 8 above)
Comment (smmribeiro) [reply pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java:107]: No, because the comparison involves some intermediate transformations.

File: pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java (hunk 7 above)
Comment (smmribeiro) [reply pentaho/src/test/java/pt/webdetails/cpf/olap/AbstractOlapUtilsTest.java:86]: No, because the comparison involves some intermediate transformations.

File: pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java (hunk 3 above)
Comment (smmribeiro) [reply pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java:45]: The JUnit infrastructure already provides the needed report when failing.
The use of try-catch-fail was due to earlier JUnit versions (and even then, it was not really necessary).

File: pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java (hunk 4 above)
Comment (smmribeiro) [reply pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java:65]: The JUnit infrastructure already provides the needed report when failing.
The use of try-catch-fail was due to earlier JUnit versions (and even then, it was not really necessary).

File: pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java (hunk 5 above)
Comment (smmribeiro) [reply pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java:84]: The JUnit infrastructure already provides the needed report when failing.
The use of try-catch-fail was due to earlier JUnit versions (and even then, it was not really necessary).

File: core/src/test/java/pt/webdetails/cpf/repository/VfsRepositoryTest.java (hunk 2 above)
Comment (smmribeiro) [reply core/src/test/java/pt/webdetails/cpf/repository/VfsRepositoryTest.java:76]: Ok, I can remove the variables and make this simpler.

File: pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java (hunk 6 above)
Comment (smmribeiro) [reply pentaho/src/test/java/pt/webdetails/cpf/MimeTypeHandlerTest.java:95]: It's no longer referred.

File: core/src/test/java/pt/webdetails/cpf/packager/MinificationPackageTest.java (hunk 1 above)
Comment (smmribeiro) [reply core/src/test/java/pt/webdetails/cpf/packager/MinificationPackageTest.java:1]: This class had no tests!

File: core/src/test/java/pt/webdetails/cpf/localization/test/PluginWriteAccess.java [hunk 13]
```diff
@@ -63,14 +63,16 @@
...
   @Override public List<IBasicFile> listFiles( String path, IBasicFileFilter filter, int maxDepth, boolean includeDirs,
                                                boolean showHiddenFilesAndFolders ) {
-    List<IBasicFile> filteredFiles = new ArrayList<IBasicFile>();
+    List<IBasicFile> filteredFiles = new ArrayList<>();
 
-    for ( IBasicFile file : storedFiles ) {
-
-      if ( filter != null && filter.accept( file ) ) {
-        filteredFiles.add( file );
+    if ( filter != null ) {
```
[the remark this answers is not in the document]
Comment (smmribeiro) [reply core/src/test/java/pt/webdetails/cpf/localization/test/PluginWriteAccess.java:68]: A filter is passed on all existing tests, so this has a slightly better performance.

File: core/src/test/java/pt/webdetails/cpf/localization/MessageBundlesHelperTest.java [hunk 14]
```diff
@@ -157,61 +145,49 @@ public void testNonAcceptedMessages() {
...
   }
 
   private List<IBasicFile> toBasicFileList( String path, String[] filesNames ) {
 
-    List<IBasicFile> basicFiles = new ArrayList<IBasicFile>();
-
-    if ( filesNames != null ) {
+    List<IBasicFile> basicFiles = new ArrayList<>();
 
+    if ( path != null && filesNames != null ) {
```
[the remark this answers is not in the document]
Comment (smmribeiro) [reply core/src/test/java/pt/webdetails/cpf/localization/MessageBundlesHelperTest.java:188]: Both path and filesNames are passed on all existing tests, so this has a slightly better performance.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 8s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **72**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.pentaho.net/job/projects/job/cpf-plugin/job/PR-168/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_
